docs/modules/ROOT/pages/references/architecture/single_sign_on.adoc

```diff
@@ -0,0 +1,215 @@
+
+Steward provides the oauth domain in the dynamic facts by reading the cluster route.
+
+[source,shell]
+----
+❯ kubectl -n openshift-authentication get route oauth-openshift -o=jsonpath='{.spec.host}'
+oauth-openshift.apps.cluster-domain.dev
+----
+
+=== Steward is extended to write a defined ConfigMap to the Lieutenant dynamic facts
+
+To allow for a way to write back static "dynamic" facts, we add a ConfigMap to the Lieutenant namespace that gets added to the dynamic facts.
+This config map is managed by `component-steward` and new facts can be added through the hierarchy.
+
+[source,yaml]
+----
+parameters:
+  steward:
+    additionalFacts:
+      vshnLdapServiceId: "{vshnLdap:serviceId}"
+----
+
+=== The controller maps LDAP groups to local client roles
+
+The controller creates client roles in Keycloak.
+It registers the client local roles with the matching groups in LDAP.
+
+The group mapping can also be manipulated in a template:
+
+[code,jsonnet]
+----
+local cluster = std.native('cluster')();
+local tenant = std.native('tenant')();
+
+local serviceGroup = '/LDAP_Customers/Service ' + if std.objectHas(cluster.status.facts, 'vshnLdapId') then
+  cluster.status.facts.vshnLdapId
+else
+  cluster.metadata.name;
+
+[
+  {
+    group: '/LDAP/VSHN openshiftroot', <1>
+    role: 'vshn-openshiftroot', <2>
+  },
+  { group: '/LDAP/VSHN openshiftrootswissonly', role: 'vshn-openshiftrootswissonly' },
+  { group: serviceGroup, role: 'customer' },
+]
+----
+<1> Keycloak group from LDAP
+<2> Keycloak client role
+
+== Example Cluster Manifest
+
+[source,yaml]
+----
+apiVersion: syn.tools/v1alpha1
+kind: Cluster
+metadata:
+  finalizers:
+  - cluster.lieutenant.syn.tools
+  - sso.appuio.io/keycloak-client <1>
+  name: c-holy-fire-9875
+  namespace: lieutenant
+  annotations:
+    oidc.sso.appuio.io/redirect-uris: '["localhost:18000","localhost:8000"]' <2>
+    sso.appuio.io/vshn-ldap-id: ClusterHolyFire9875 <3>
+spec:
+  displayName: Cybertron Prod 1
+  facts:
+    distribution: openshift4 <4>
+  [...]
+  tenantRef:
+    name: t-frosty-forest-1224 <5>
+status:
+  facts:
+    kubernetesVersion: '{"buildDate":"2023-09-11T02:22:18Z","compiler":"gc","gitCommit":"f10a517f7199bdae922a70893d85eb96a76f5c2d","gitTreeState":"clean","gitVersion":"v1.26.7+c7ee51f","goVersion":"go1.19.10
+      X:strictfipsruntime","major":"1","minor":"26","platform":"linux/amd64"}'
+    openshiftVersion: '{"Major":"4","Minor":"13","Patch":"13"}'
+    oauthDomain: https://oauth-openshift.apps.c-holy-fire-9875.dev <6>
+    vshnLdapServiceId: ClusterHolyFire9875 <7>
+----
+<1> The `sso.appuio.io/keycloak-client` finalizer is added to the cluster object to allow cleanup of the Keycloak client when the cluster is deleted.
+<2> The `oidc.sso.appuio.io/redirect-uris` annotation is used to override the default redirect uris.
+<3> The `sso.appuio.io/vshn-ldap-id` annotation is used to override the default LDAP group mapping.
```

Comment:
There's an argument here that `sso.appuio.io` isn't the right domain for any of the annotations.

I'd argue that the SSO controller itself is useful for Project Syn in general, and therefore the first two annotations could be prefixed with `sso.syn.tools` and `oidc.sso.syn.tools`, while the last annotation is so VSHN-specific that it really should be `sso.vshn.net/ldap-id` or similar.

Reply:
True. Much happier with your proposed naming.